File: check_braces.js

const fs = require('fs');
const content = fs.readFileSync('c:\\Users\\Baha\\Desktop\\NEXCHAT\\chat.js', 'utf8');

let stack = [];
let lines = content.split('\n');

for (let i = 0; i < lines.length; i++) {
    let line = lines[i];
    for (let j = 0; j < line.length; j++) {
        let char = line[j];
        if (char === '{') {
            stack.push({ line: i + 1, char: j + 1 });
        } else if (char === '}') {
            if (stack.length === 0) {
                console.log(`Extra '}' at line ${i + 1}, column ${j + 1}`);
            } else {
                stack.pop();
            }
        }
    }
}

if (stack.length > 0) {
    stack.forEach(unclosed => {
        console.log(`Unclosed '{' at line ${unclosed.line}, column ${unclosed.char}`);
    });
} else {
    console.log("Braces are balanced (simple check)");
}

File: check_braces.py

import os

filepath = r'c:\Users\Baha\Desktop\NEXCHAT\chat.js'
with open(filepath, 'r', encoding='utf-8') as f:
    content = f.read()

stack = []
for i, line in enumerate(content.split('\n'), 1):
    for j, char in enumerate(line, 1):
        if char == '{':
            stack.append((i, j))
        elif char == '}':
            if not stack:
                print(f"Extra '}}' at line {i}, column {j}")
            else:
                stack.pop()

for line, col in stack:
    print(f"Unclosed '{{' at line {line}, column {col}")

if not stack:
    print("Braces are balanced (simple check)")
